File: app/src/main/java/org/singularux/contacts/core/ContactsPermissionManager.java
package org.singularux.contacts.core;

import androidx.annotation.NonNull;

import java.util.Arrays;

public interface ContactsPermissionManager {

    boolean hasPermission(@NonNull ContactsPermission permission);

    default boolean hasAllPermissions(@NonNull ContactsPermission... permissions) {
        return Arrays.stream(permissions).allMatch(this::hasPermission);
    }

    @NonNull String getPermissionString(@NonNull ContactsPermission permission);

    default @NonNull String[] getPermissionStrings(@NonNull ContactsPermission... permissions) {
        return Arrays.stream(permissions)
                .map(this::getPermissionString)
                .toArray(String[]::new);
    }

}

File: core/permission/src/main/kotlin/org/singularux/contacts/core/permission/ContactsPermissionManager.kt
package org.singularux.contacts.core.permission

interface ContactsPermissionManager {

    fun hasPermission(permission: ContactsPermission): Boolean
    fun getPermissionString(permission: ContactsPermission): String

    fun hasPermissions(vararg permissions: ContactsPermission): Boolean {
        return permissions.map { hasPermission(it) }.all { it }
    }

    fun getPermissionsString(vararg permissions: ContactsPermission): List<String> {
        return permissions.map { getPermissionString(it) }
    }

}